server.go

```diff
@@ -609,53 +613,45 @@ func (s *Server) Shards(database string) ([]*Shard, error) {
 		return nil, ErrDatabaseNotFound
 	}
 
-	// Retrieve shards from database.
-	shards := make([]*Shard, 0, len(db.shards))
-	for _, shard := range db.shards {
-		shards = append(shards, shard)
-	}
-	return shards, nil
-}
-
-// shardsByTimestamp returns all shards that own a given timestamp for a database.
-func (s *Server) shardsByTimestamp(database, policy string, timestamp time.Time) ([]*Shard, error) {
-	db := s.databases[database]
-	if db == nil {
-		return nil, ErrDatabaseNotFound
+	// Retrieve groups from database.
+	var a []*ShardGroup
+	for _, rp := range db.policies {
+		for _, g := range rp.groups {
+			a = append(a, g)
+		}
 	}
-	return db.shardsByTimestamp(policy, timestamp)
+	return a, nil
 }
 
-// CreateShardsIfNotExist creates all the shards for a retention policy for the interval a timestamp falls into.
-// Note that multiple shards can be created for each bucket of time.
-func (s *Server) CreateShardsIfNotExists(database, policy string, timestamp time.Time) error {
-	c := &createShardIfNotExistsCommand{Database: database, Policy: policy, Timestamp: timestamp}
-	_, err := s.broadcast(createShardIfNotExistsMessageType, c)
+// CreateShardGroupIfNotExist creates the shard group for a retention policy for the interval a timestamp falls into.
+func (s *Server) CreateShardGroupIfNotExists(database, policy string, timestamp time.Time) error {
+	c := &createShardGroupIfNotExistsCommand{Database: database, Policy: policy, Timestamp: timestamp}
+	_, err := s.broadcast(createShardGroupIfNotExistsMessageType, c)
 	return err
 }
 
 // createShardIfNotExists returns the shard for a given retention policy, series, and timestamp.
 // If it doesn't exist, it will create all shards for the given timestamp
-func (s *Server) createShardIfNotExists(database, policy string, id uint32, timestamp time.Time) (*Shard, error) {
-	// Check if shard exists first.
-	sh, err := s.shardByTimestamp(database, policy, id, timestamp)
+func (s *Server) createShardGroupIfNotExists(database, policy string, timestamp time.Time) (*ShardGroup, error) {
```

Comment:
Yes, it previously created "shards" instead of "shard group" and looked up the shard by series id. I extracted the shard lookup from the function so it just does group creation.

Fixed the comment: fab5082